### PR TITLE
updated main title section with 2023 info

### LINKED ISSUE #2: Update main title section with 2023 info
- [x] Replace HackNC 2021 with 2023
- [x] Add updated dates (October 27-29th)
- [x] Remove "event ended" text
- [x] Remove "(this year, we'll be having a hybrid hackathon!)" from the Who We Are section
- [x] Remove hacker interest form button and change other button to "2023 Organizer Interest form"
- [x] Add link to organizer form: https://docs.google.com/forms/d/e/1FAIpQLSdobyqr0nq4ZXEZi0pdTGTBIuOZeEr5B26Wz71sij77ds15Xw/viewform

### PR DESCRIPTION
**Pull request by aryasalian**
This resolves task #2 from the Projects tab. I have also updated the outdated info in the footer(the "Looking to sponsor HackNC 2021? Email us at [sponsorship@hacknc.com](mailto:sponsorship@hacknc.com)" text and the copyright "HackNC 2021 – "). Also updated the meta and title info of the website.

### CODE REVIEW

File: index.html
Comment (atoneyd) [inline index.html]: Can you remove the dash at the end? not sure why it was there before

**Review by atoneyd — APPROVED** [review]
LGTM! Just had a small edit, once fixed you can merge this branch.

File: index.html
Comment (aryasalian) [reply index.html]: alright, on it!

**Pull request merged**
